components/api/src/main/java/com/hotels/styx/api/HttpRequest.java

```diff
@@ -183,21 +217,45 @@ public static Builder patch(String uri, StyxObservable<ByteBuf> body) {
         return new Builder(PATCH, uri).body(body);
     }
 
+    /**
+     * Returns the HTTP protocol version.
+     * @return
+     */
     @Override
     public HttpVersion version() {
         return this.version;
     }
 
+    /**
+     * Returns all HTTP headers as {@link HttpHeaders} instance.
+     *
+     * @return {@link HttpHeaders} object.
+     */
     @Override
     public HttpHeaders headers() {
         return headers;
     }
 
+    /**
+     * Returns all values for a given HTTP header name.
+     *
+     * Returns an empty list if header is not present.
+     *
+     * @param name header name
+     * @return A list of all header names.
```

Comment:
In Javadoc convention, the text after a tag does not start with a capital letter. Presumably, this is because the text is not a full sentence.

Reply:
Same comment applies to the rest.

Reply:
FYI:

From: http://www.oracle.com/technetwork/java/javase/documentation/index-137868.html#@param.

> The description begins with a lowercase letter if it is a phrase (contains no verb), or an uppercase letter if it is a sentence. End the phrase with a period only if another phrase or sentence follows it.

Clearly in this case it is a phrase instead of a sentence.